editor/modes/visual-editor/block.js

```diff
@@ -340,8 +340,10 @@ class VisualEditorBlock extends Component {
 		}
 
 		// Generate a class name for the block's editable form
-		let { className = getBlockDefaultClassname( block.name ) } = blockType;
-		className = classnames( className, block.attributes.className );
+		const generatedClassName = hasBlockSupport( blockType, 'generatedClassName', true ) ?
```

Comment:
If generated class name is managed by a hook, should the visual editor have any awareness of it at all? Can this be extended as well?

Reply:
I think it can not because it's not applied to the root level systematically like we do in `save` :( which bring us to dropping this entirely and just passing the prop in `save` like we do in `edit`